### PR TITLE
Move remaining Consensus specific files to Consensus project

### PR DESCRIPTION
**Pull request by quantumagi**
This PR moves the remaining Consensus-specific files to the Consensus project

### CODE REVIEW

File: Stratis.Bitcoin.Features.RPC/RpcSettings.cs [hunk 1]
```diff
@@ -58,58 +58,69 @@ public void Load(NodeSettings nodeSettings)
...
-                this.Bind = config
-                    .GetAll("rpcbind")
-                    .Select(p => NodeSettings.ConvertToEndpoint(p, defaultPort))
-                    .ToList();
-            }
-            catch (FormatException)
-            {
-                throw new ConfigurationException("Invalid rpcbind value");
-            }
+            this.Server = true;
```
Comment (bokobza) [inline Stratis.Bitcoin.Features.RPC/RpcSettings.cs]: why do we set the server to true here?

File: Stratis.Bitcoin.Features.RPC/RpcSettings.cs [hunk 2]
```diff
@@ -58,58 +58,69 @@ public void Load(NodeSettings nodeSettings)
...
-            if (!this.Server)
-                return;
+            this.RPCPort = nodeSettings.Network.RPCPort;
+
+            if (this.Server)
+            {
+                this.RpcUser = config.GetOrDefault<string>("rpcuser", null);
+                this.RpcPassword = config.GetOrDefault<string>("rpcpassword", null);
+
+                var defaultPort = config.GetOrDefault<int>("rpcport", nodeSettings.Network.RPCPort);
```
Comment (bokobza) [inline Stratis.Bitcoin.Features.RPC/RpcSettings.cs]: You can remove the variable 'defaultPort' here.

File: Stratis.Bitcoin.Features.RPC/RpcSettings.cs [hunk 3]
```diff
@@ -58,58 +58,69 @@ public void Load(NodeSettings nodeSettings)
...
+                        .Select(p => IPAddress.Parse(p))
+                        .ToList();
+                }
+                catch (FormatException)
+                {
+                    throw new ConfigurationException("Invalid rpcallowip value");
+                }
+            }
+
+            this.callback?.Invoke(this);
```
Comment (bokobza) [inline Stratis.Bitcoin.Features.RPC/RpcSettings.cs]: I find the fact that we have this callback invoked in the middle of the method quite confusing.
Can we do something like this:
1. get values from config
2. invoke callback
3. check validity of settings

File: Stratis.Bitcoin.Features.RPC/RpcSettings.cs (hunk 2 above)
Comment (quantumagi) [reply Stratis.Bitcoin.Features.RPC/RpcSettings.cs]: Done

File: Stratis.Bitcoin.Features.RPC/RpcSettings.cs (hunk 3 above)
Comment (quantumagi) [reply Stratis.Bitcoin.Features.RPC/RpcSettings.cs]: I agree. I've re-factored the code by splitting it into additional private methods.

File: Stratis.Bitcoin.Features.RPC/RpcSettings.cs (hunk 1 above)
Comment (quantumagi) [reply Stratis.Bitcoin.Features.RPC/RpcSettings.cs]: Due to the preceding check the server variable would already have a value of true at that point - unless the rpc user and password were set via callback (in the typical "addRpc" scenario), in which case we would want the rpc server to be active.

**Comment by bokobza** [comment]
Thanks G.

**Pull request merged**